### ISSUE TITLE
Strange progress

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#11819]
When I do a full index, the list of tasks is shown as complete, but there is some unnamed (!) task running:

![image](https://github.com/user-attachments/assets/88e401f2-093e-4205-a865-478f39b1847b)

Maybe, this is related to https://github.com/JabRef/jabref/issues/11701

**Comment by Siedlerchr** [comment]
Related code from TaskProgressView:

```
   EventHandler<WorkerStateEvent> taskHandler = evt -> {
            if (!isRetainTasks()) {
                if (evt.getEventType().equals(
                        WorkerStateEvent.WORKER_STATE_SUCCEEDED)
                        || evt.getEventType().equals(
                        WorkerStateEvent.WORKER_STATE_CANCELLED)
                        || evt.getEventType().equals(
                        WorkerStateEvent.WORKER_STATE_FAILED)) {
                    getTasks().remove(evt.getSource());
                }
            }
        };
```

**Comment by LoayGhreeb** [comment]
Oh, this issue is fixed by #11822. I figured out that I used a wrong link in the PR.
